ekncontent/ekncontent/eknc-content-object-model.c

```diff
@@ -0,0 +1,580 @@
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:thumbnail-uri:
+   *
+   * URI of the the thumbnail image
+   *
+   * Usually the EKN ID of an #ImageObjectModel.
+   */
+  eknc_content_object_model_props[PROP_THUMBNAIL_URI] =
+    g_param_spec_string ("thumbnail-uri", "Thumbnail URI",
+       "URI of the thumbnail image",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:language:
+   *
+   * The language for this content object. Defaults to an empty string.
+   */
+  eknc_content_object_model_props[PROP_LANGUAGE] =
+    g_param_spec_string ("language", "Language",
+      "Language of the document or media object",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:copyright-holder:
+   *
+   * The copyright holder for this content object. Defaults to an empty string.
+   */
+  eknc_content_object_model_props[PROP_COPYRIGHT_HOLDER] =
+    g_param_spec_string ("copyright-holder", "Copyright Holder",
+      "The copyright holder of the object",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:source-uri:
+   *
+   * URI where this content was downloaded from during database build
+   *
+   * TODO Should we check to always have a value for source-uri?
+   * Don"t use this property for user-visible things.
+   * It is only used internally.
+   */
+  eknc_content_object_model_props[PROP_SOURCE_URI] =
+    g_param_spec_string ("source-uri", "Source URL",
+      "URI where this content was downloaded from during database build",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:content-type:
+   *
+   * The source content"s mimetype
+   */
+  eknc_content_object_model_props[PROP_CONTENT_TYPE] =
+    g_param_spec_string ("content-type", "Object Content Type",
+      "Mimetype of the source content",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:synopsis:
+   *
+   * The synopsis for this content object. Defaults to an empty string.
+   */
+  eknc_content_object_model_props[PROP_SYNOPSIS] =
+    g_param_spec_string ("synopsis", "Synopsis",
+      "Synopsis of the document or media object",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:last-modified-date:
+   *
+   * The date of last modification for this content object. It treats dates
+   * according to the ISO8601 standard.
+   */
+  eknc_content_object_model_props[PROP_LAST_MODIFIED_DATE] =
+    g_param_spec_string ("last-modified-date", "Last Modified Date",
+      "Last Modified Date of the document or media object",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
```

Comment:
Is there a boxed type for dates in GLib we could use here? Maybe with an override to convert to/from `Date` objects in JS?